enterprise/web.jsf.editor/src/org/netbeans/modules/web/jsf/editor/JsfSupportImpl.java

```diff
@@ -58,6 +61,21 @@ public class JsfSupportImpl implements JsfSupport {
 
 	private static final Logger LOG = Logger.getLogger(JsfSupportImpl.class.getSimpleName());
 
+    private static final Map<JSFVersion, JsfVersion> JSF_VERSION_MAPPING;
+    static {
+        Map<JSFVersion, JsfVersion> map = new HashMap<>();
+        map.put(JSFVersion.JSF_1_0, JsfVersion.JSF_1_0);
+        map.put(JSFVersion.JSF_1_1, JsfVersion.JSF_1_1);
+        map.put(JSFVersion.JSF_1_2, JsfVersion.JSF_1_2);
+        map.put(JSFVersion.JSF_2_0, JsfVersion.JSF_2_0);
+        map.put(JSFVersion.JSF_2_1, JsfVersion.JSF_2_1);
+        map.put(JSFVersion.JSF_2_2, JsfVersion.JSF_2_2);
+        map.put(JSFVersion.JSF_2_3, JsfVersion.JSF_2_3);
+        map.put(JSFVersion.JSF_3_0, JsfVersion.JSF_3_0);
+        map.put(JSFVersion.JSF_4_0, JsfVersion.JSF_4_0);
+        JSF_VERSION_MAPPING = Collections.unmodifiableMap(map);
```

Comment:
Is there a reason for the two version implementations?

Reply:
I'm not happy about that mapping as well.

Basically `JSFVersion` is not part of the api module but of implementation module `web.jsf`. So reusing `JSFVersion`  would have meant either to:

* Move it to `web.jsfapi`
or
* Add `web.jsf` as dependency

From my perspective it makes more sense to have it as part of the api, at least the feature set which is currently implemented. On the other hand I did not want to merge both classes as it would have created an even huger patchset than already.
So basically the idea would be to eliminate `JSFVersion` and integrate it to `JsfVersion` or somewhere else.